src/NServiceBus.Core/Saga/ISagaPersister.cs

```diff
@@ -10,39 +10,46 @@ public interface ISagaPersister
     {
 		/// <summary>
 		/// Saves the saga entity to the persistence store.
-		/// </summary>
+        /// </summary>
+        /// <param name="metadata">Metadata describing the saga being saved.</param>
 		/// <param name="saga">The saga entity to save.</param>
-        void Save(IContainSagaData saga);
+        void Save(SagaMetadata metadata, IContainSagaData saga);
```

Comment:
I know obsoleting interfaces is practically not possible. So we would require to introduce a completely new interface for the changes and obsolete the old one. Is it worth the hassle? Don't know. What is your thought @SimonCropp ?